src/wazuh_db/wdb_agents.c

```diff
@@ -1,18 +1,108 @@
 #include "wdb_agents.h"
 
-int wdb_agents_insert_vuln_cve(wdb_t *wdb, const char* name, const char* version, const char* architecture, const char* cve) {
+int wdb_agents_find_package(wdb_t *wdb, const char* reference){
+    sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_PROGRAM_FIND);
 
-    sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_VULN_CVE_INSERT);
     if (stmt == NULL) {
         return OS_INVALID;
     }
 
-    sqlite3_bind_text(stmt, 1, name, -1, NULL);
-    sqlite3_bind_text(stmt, 2, version, -1, NULL);
-    sqlite3_bind_text(stmt, 3, architecture, -1, NULL);
-    sqlite3_bind_text(stmt, 4, cve, -1, NULL);
+    sqlite3_bind_text(stmt, 1, reference, -1, NULL);
 
-    return wdb_exec_stmt_silent(stmt);
+    switch (sqlite3_step(stmt)) {
+    case SQLITE_ROW:
+        return 1;
+        break;
+    case SQLITE_DONE:
+        return 0;
+        break;
+    default:
+        mdebug1("DB(%s) sqlite3_step(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+        return -1;
+    }
+}
+
+int wdb_agents_find_cve(wdb_t *wdb, const char* cve, const char* reference){
+    sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_VULN_CVE_FIND_CVE);
+
+    if (stmt == NULL) {
+        return OS_INVALID;
+    }
+
+    sqlite3_bind_text(stmt, 1, cve, -1, NULL);
+    sqlite3_bind_text(stmt, 2, reference, -1, NULL);
+
+    switch (sqlite3_step(stmt)) {
```

Comment:
We can use **wdb_exec_stmt_silent()** that wraps the execution of a statement that doesn't need a payload as response.

Reply:
This wrapper method returns OS_SUCCESS (0) for both, SQLITE_ROW and SQLITE_DONE. Is it required to only consider the SQLITE_ROW as a valid answer?
In that case, we should make **wdb_agents_find_cve()** return value a boolean, to only respond if the package was found or not.